subprojects/gradle/build-checks/src/main/kotlin/com/avito/android/build_checks/AndroidAppChecksExtension.kt

```diff
@@ -38,28 +39,30 @@ public open class AndroidAppChecksExtension : BuildChecksExtension() {
              */
             public val ignoredResourceTypes: MutableList<String> = mutableListOf()
 
+            internal val ignoredResources: MutableSet<Resource> = mutableSetOf()
+
             /**
              * Key is a resource type: layout, string, integer, ... as it used in XML.
              * See all names in [com.android.resources.ResourceType]
              *
              * Value: resource name
              */
-            public val ignoredResources: MutableMap<String, String> = mutableMapOf()
+            public fun ignoredResource(type: String, name: String) {
```

Comment:
Propose change type to `enum` or add validation of `type`

Reply:
I've chosen strings to avoid knowledge about new abstractions in build scripts. 

```kotlin
uniqueAppResources {
   ignoredResource("string", "title")
}
```

It's copying what user see in resources files:

```xml
<string name="title">Main</string>
```

Values are validated in this function.